High intraspecies allelic diversity in Arabidopsis NLR immune receptors is associated with distinct genomic and epigenomic features
====
[![Preprint DOI](https://img.shields.io/badge/Preprint_DOI-10.1101/2023.01.12.523861-red)](https://doi.org/10.1101/2023.01.12.523861)
[![Zenodo DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.8411396.svg)](https://doi.org/10.5281/zenodo.8411396)

This repository contains the code for the manuscript, "High intraspecies allelic diversity in Arabidopsis NLR immune receptors is associated with distinct genomic and epigenomic features," currently pre-printed on bioRxiv. The repository is organized such that each directory contains one type of analysis described in the manuscript containing scripts run on UC Berkeley's High Performance Computing Cluster, any associated individual job scripts, and post processing. All figure generation and statistical tests are in the `figure_generation` directory. All files used as inputs for figure generation are available on Zenodo.  

Authors
-----
* **Chandler A. Sutherland**
<a itemprop="sameAs" content="https://orcid.org/0000-0001-5840-7661" href="https://orcid.org/0000-0001-5840-7661" target="orcid.widget" rel="me noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon"></a>  
*Department of Plant and Microbial Biology, University of California - Berkeley*

* **Daniil M. Prigozhin**
<a itemprop="sameAs" content="https://orcid.org/0000-0003-2075-0231" href="https://orcid.org/0000-0003-2075-0231" target="orcid.widget" rel="me noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon"></a>  
*Molecular Biophysics and Integrated Bioimaging Division, Lawrence Berkeley National Laboratory*

* **J. Grey Monroe**
<a itemprop="sameAs" content="https://orcid.org/0000-0002-4025-5572" href="https://orcid.org/0000-0002-4025-5572" target="orcid.widget" rel="me noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon"></a>  
*Department of Plant Sciences, University of California - Davis*

* **Ksenia V. Krasileva**
<a itemprop="sameAs" content="https://orcid.org/0000-0002-1679-0700" href="https://orcid.org/0000-0002-1679-0700" target="orcid.widget" rel="me noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon"></a>[:e-mail:](mailto:kseniak@berkeley.edu)  
*Department of Plant and Microbial Biology, University of California - Berkeley*

Abstract 
----
>Plants rely on Nucleotide-binding, Leucine-rich repeat Receptors (NLRs) for pathogen recognition. Highly variable NLRs (hvNLRs) show remarkable intraspecies diversity, while their low variability paralogs (non-hvNLRs) are conserved between ecotypes. At a population level, hvNLRs provide new pathogen recognition specificities, but the association between allelic diversity and genomic and epigenomic features has not been established. Our investigation of NLRs in Arabidopsis Col-0 has revealed that hvNLRs show higher expression, less gene body cytosine methylation, and closer proximity to transposable elements than non-hvNLRs. hvNLRs show elevated synonymous and nonsynonymous nucleotide diversity and are in chromatin states associated with an increased probability of mutation. Diversifying selection maintains variability at a subset of codons of hvNLRs, while purifying selection maintains conservation at non-hvNLRs. How these features are established and maintained, and whether they contribute to the observed diversity of hvNLRs is key to understanding the evolution of plant innate immune receptors.

License 
----
Code is freely available under the MIT license  

Contact 
----
Please feel free to contact me directly with any questions or issues with the code  
Chandler Sutherland - [@chandler_suth](https://twitter.com/chandler_suth) - chandlersutherland@berkeley.edu
