{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7efcb754",
   "metadata": {},
   "outputs": [],
   "source": [
    "import egglib\n",
    "import glob \n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ba25997",
   "metadata": {},
   "source": [
    "Goal: calculate nucleotide diversity per site and Tajima's D for NLR clades using the egglib python package "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8db0298f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#gather alignments in fasta format \n",
    "directory='/global/scratch/users/chandlersutherland/e14/popgen/popgenome_test'\n",
    "files = glob.glob(os.path.join(directory, \"*.fas\"))\n",
    "#import nlr annotations generated by nlr_annotation.ipynb \n",
    "annotations=pd.read_csv('/global/scratch/users/chandlersutherland/e14/popgen/nlr_aa_annotation.csv')[['clade', 'start', 'end', 'domain']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3aa54bb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#write a function that takes in protein alignment coordinates and clade and outputs the desired stats \n",
    "def sub_stat(row):\n",
    "    #load alignment, and subset by domain\n",
    "    fasta='/global/scratch/users/chandlersutherland/e14/popgen/popgenome_test/'+row[0]+'.pal2nal.fas'\n",
    "    clade=row[0]\n",
    "    domain=row[3]\n",
    "    aln=egglib.io.from_fasta(fasta, alphabet=egglib.alphabets.DNA)\n",
    "    codon=egglib.tools.to_codons(aln)\n",
    "    subset=codon.extract(int(row[1]), int(row[2]))\n",
    "    \n",
    "    #compute across subset stats\n",
    "    dna=egglib.tools.to_bases(subset)\n",
    "    ls=egglib.stats.ComputeStats()\n",
    "    ls.configure(multi_hits=True, multi=False)\n",
    "    ls.add_stats('Pi', 'lseff', 'D')\n",
    "    stats=ls.process_align(dna, max_missing=0.1)\n",
    "    \n",
    "    lseff=stats['lseff']\n",
    "    D=stats['D']\n",
    "    Pi=stats['Pi']\n",
    "    dna_aln=dna.ls\n",
    "    \n",
    "    #compute per synonymous and nonsynonomous site stats \n",
    "    cs_subset = egglib.stats.CodingDiversity(subset)\n",
    "\n",
    "    #use the process_sites function to calculate Pi over just the synonymous and non-synonymous sites \n",
    "    statsS=ls.process_sites(cs_subset.sites_S)\n",
    "    statsNS=ls.process_sites(cs_subset.sites_NS) \n",
    "\n",
    "    #normalize by the number of synonymous and nonsynonymous sites\n",
    "    #if there are no polymorphisms at each site, return NaN\n",
    "    \n",
    "    if cs_subset.num_pol_S == 0:\n",
    "        PiS=np.nan\n",
    "    else:\n",
    "        PiS=statsS['Pi']/cs_subset.num_sites_S\n",
    "    \n",
    "    if cs_subset.num_pol_NS == 0:\n",
    "        PiNS=np.nan\n",
    "    else:\n",
    "        PiNS=statsNS['Pi']/cs_subset.num_sites_NS\n",
    "    \n",
    "    return({'clade':clade, 'domain':domain, 'Pi_raw':Pi, 'lseff':lseff, 'dna_aln':dna_aln, 'PiS':PiS, 'PiN':PiNS, 'D':D})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bc8ef31",
   "metadata": {},
   "outputs": [],
   "source": [
    "stats=pd.DataFrame()\n",
    "#calculate per domain stats \n",
    "for i in range(0, len(annotations)):\n",
    "    row=annotations.iloc[i,:]\n",
    "    result=sub_stat(row)\n",
    "    result['start']=row[1]\n",
    "    result['stop']=row[2]\n",
    "    #print(result)\n",
    "    stats=stats.append(result, ignore_index=True)\n",
    "\n",
    "stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eae24a7a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#add Col-0 gene names and HV label\n",
    "gene_table=pd.read_table('/global/scratch/users/chandlersutherland/e14/popgen/Atha_NLRome_GeneTable.txt')[['Gene', 'Clade', 'Ecotype', 'HV']]\n",
    "gene_table=gene_table[gene_table['Ecotype']=='ATHALIANA']\n",
    "gene_table['Gene']=gene_table['Gene'].str.split('_', expand=True)[1]\n",
    "gene_table=gene_table.rename(columns={'Clade':'clade'})[['Gene', 'clade', 'HV']]\n",
    "gene_table=gene_table.assign(HV=gene_table.HV.map({0:'non-hv', 1:'hv'}))\n",
    "\n",
    "egglib_results=pd.merge(stats, gene_table)\n",
    "egglib_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c9b1c81",
   "metadata": {},
   "outputs": [],
   "source": [
    "egglib_results.to_csv('/global/scratch/users/chandlersutherland/e14/popgen/egglib_results.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62db8706",
   "metadata": {},
   "source": [
    "Complement the per domain analysis with sliding window"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9ac7dcc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_only=annotations[annotations['domain']=='cds']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ad31afe",
   "metadata": {},
   "outputs": [],
   "source": [
    "#function that takes in a clade name, and evaluates Pi, PiS, PiNS, and D across sliding windows of window size 33 codons and step size 10 codons \n",
    "#outputs a dataframe with the results across the codon windows \n",
    "def my_slider(row, window, step):\n",
    "    #get codon length \n",
    "    codon_length=row[2]\n",
    "    clade=row[0]\n",
    "    \n",
    "    #generate a new annotations dataframe with sliding windows to this length \n",
    "    fake_annotations=pd.DataFrame()\n",
    "    for i in range(0, int(codon_length/step)):\n",
    "        fake_annotations=fake_annotations.append({'start':0+i*step, 'end':window+i*step, 'clade':clade, 'domain':'window'}, ignore_index=True)\n",
    "    fake_annotations=fake_annotations[['clade', 'start', 'end', 'domain']]\n",
    "    return fake_annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6aeb0cab",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create new annotations df with sliding windows\n",
    "slide=[]\n",
    "for i in range(0, len(cds_only)):\n",
    "    row=cds_only.iloc[i,:]\n",
    "    windows=my_slider(row, 100, 25)\n",
    "    slide.append(windows)\n",
    "\n",
    "slide_df=pd.concat(slide)\n",
    "slide_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "153c8e59",
   "metadata": {},
   "outputs": [],
   "source": [
    "window_stats=pd.DataFrame()\n",
    "for i in range(0, len(slide_df)):\n",
    "    row=slide_df.iloc[i,:]\n",
    "    result=sub_stat(row)\n",
    "    result['start']=row[1]\n",
    "    result['stop']=row[2]\n",
    "    #print(result)\n",
    "    window_stats=window_stats.append(result, ignore_index=True)\n",
    "\n",
    "window_stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fae5fac",
   "metadata": {},
   "outputs": [],
   "source": [
    "vignette=slide_df[slide_df['clade'].isin(['Int8443_258_351_L_88_120_L_38','Int8443_258_351_L_88_120_R_50'])]\n",
    "for i in range(0, len(vignette)):\n",
    "    row=slide_df.iloc[i,:]\n",
    "    result=sub_stat(row)\n",
    "    result['start']=row[1]\n",
    "    result['stop']=row[2]\n",
    "    #print(result)\n",
    "    window_stats=window_stats.append(result, ignore_index=True)\n",
    "\n",
    "window_stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20b0f72c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vignette_window_results=pd.merge(window_stats, gene_table)\n",
    "vignette_window_results\n",
    "vignette_window_results.to_csv('/global/scratch/users/chandlersutherland/e14/popgen/egglib_vignette_window_stats.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33cfad34",
   "metadata": {},
   "outputs": [],
   "source": [
    "egglib_window_final=pd.merge(egglib_window_results, gene_table)\n",
    "egglib_window_final.to_csv('/global/scratch/users/chandlersutherland/e14/popgen/egglib_window_stats.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
