{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30a794f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import pandas as pd\n",
    "import sys\n",
    "import glob\n",
    "import numpy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0b59ce8",
   "metadata": {},
   "source": [
    "The goal of this notebook is to collate desired sequences for each clade in a directory titled by that clade. Have functions for exons, transcripts, and gene sequences. Welcome to the roundup! &#x1F920;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74f18796",
   "metadata": {},
   "outputs": [],
   "source": [
    "#first, find the files associated with final clade membership by using the gene table \n",
    "base='/global/scratch/users/chandlersutherland/e14/NLRCladeFinder/Atha_NLRome/'\n",
    "gene_table=base+\"Atha_NLRome_GeneTable.txt\"\n",
    "gene=pd.read_csv(gene_table,sep = '\\t')\n",
    "gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17d9a156",
   "metadata": {},
   "outputs": [],
   "source": [
    "#generate a dictionary with Clade names as the keys and representative genes as the values by getting each gene in the table associated with the clade\n",
    "dictionary=dict()\n",
    "clades=numpy.unique(gene['Clade'])\n",
    "for clade in clades:\n",
    "    print(clade)\n",
    "    genes=list(gene[gene['Clade']==clade]['Gene'])\n",
    "    dictionary[clade]=genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d77743b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define CDS roundup, which takes the items of this dictionary and writes a fasta file with all exon sequences to the clade folder\n",
    "def CDS_roundup(clade, gene_list):\n",
    "    #initialize some housekeeping, count being the number of found exons and found a set of gene names \n",
    "    count=0\n",
    "    found=[]\n",
    "    \n",
    "    #get the clade name and initialize the new output directory \n",
    "    output_dir=\"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    if not os.path.exists(output_dir): #this prevents a directory exists error\n",
    "       os.makedirs(output_dir)\n",
    "    \n",
    "    #read in the gene list per clade, and clean up \n",
    "    clean_list=[]\n",
    "    for i in gene_list:\n",
    "        clean=i.replace('_1', '.1').replace('_2', '.2').replace('_3', '.3')\n",
    "        clean_list.append(clean)\n",
    "\n",
    "    # create output CDS file \n",
    "    g = open(output_dir+\"/\"+clade+'.CDS.fa', \"w\")\n",
    "\n",
    "    # open the NLRome CDS file and search for gene names \n",
    "    #if found, write to new fasta file \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/hvNLR/ArabidopsisRENSEQ/CDS/NLRome.CDS.fasta\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "            for gene in clean_list:\n",
    "                if gene in line:\n",
    "                    count=count+1\n",
    "                    found.append(gene)\n",
    "                    g.writelines(line)\n",
    "                    g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #special treatment for Col-0 because of the full gene names \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/popgen/Col0_Exons.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "                for gene in clean_list:\n",
    "                    if gene.startswith('ATHALIANA'):\n",
    "                        short=gene.replace('ATHALIANA_', '')\n",
    "                        if short in line: \n",
    "                            count=count+1\n",
    "                            found.append(gene)\n",
    "                            g.writelines(line)\n",
    "                            g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #sanity check. The found genes should be the same as the input genes. \n",
    "    print('found '+str(count)+' transcripts')\n",
    "    s=set(found)\n",
    "    q=set(list(clean_list))\n",
    "    if s == q:\n",
    "          print('found all genes supplied')\n",
    "    else:\n",
    "          print('error, some genes missing')\n",
    "          temp3 = [x for x in q if x not in s]\n",
    "          print(str(temp3)+' not found')\n",
    "    \n",
    "    g.close()\n",
    "    print('CDS file written to '+output_dir+\"/\"+clade+'.CDS.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59d4b868",
   "metadata": {},
   "outputs": [],
   "source": [
    "for k, v in dictionary.items():\n",
    "    CDS_roundup(k, v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82974e0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define Gene roundup, which takes the items of this dictionary and writes a fasta file with all Gene sequences to the clade folder\n",
    "def Gene_roundup(clade, gene_list):\n",
    "    #initialize some housekeeping, count being the number of found exons and found a set of gene names \n",
    "    count=0\n",
    "    found=[]\n",
    "    \n",
    "    #get the clade name and initialize the new output directory \n",
    "    output_dir=\"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    if not os.path.exists(output_dir): #this prevents a directory exists error\n",
    "       os.makedirs(output_dir)\n",
    "    \n",
    "    #read in the gene list per clade, and clean up \n",
    "    clean_list=[]\n",
    "    for i in gene_list:\n",
    "        clean=i.replace('_T', '_G').replace('-R1', '').replace('_1', '.1').replace('_2', '.2').replace('_3', '.3')\n",
    "        clean_list.append(clean)\n",
    "\n",
    "    # create output CDS file \n",
    "    g = open(output_dir+\"/\"+clade+'.GeneSequence_test.fa', \"w\")\n",
    "\n",
    "    # open the NLRome CDS file and search for gene names \n",
    "    #if found, write to new fasta file \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/hvNLR/ArabidopsisRENSEQ/GeneSequences/NLRome_Gene_Sequences.respectify.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "            for gene in clean_list:\n",
    "                if gene in line:\n",
    "                    count=count+1\n",
    "                    found.append(gene)\n",
    "                    g.writelines(line)\n",
    "                    g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #special treatment for Col-0 because of the full gene names \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/popgen/Col0_GeneSequences.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "                for gene in clean_list:\n",
    "                    if gene.startswith('ATHALIANA'):\n",
    "                        short=gene.replace('ATHALIANA_', '').replace('_1', '').replace('.1', '')\n",
    "                        if short in line: \n",
    "                            count=count+1\n",
    "                            found.append(gene)\n",
    "                            g.writelines(line)\n",
    "                            g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #sanity check. The found genes should be the same as the input genes. \n",
    "    print('found '+str(count)+' genes')\n",
    "    s=set(found)\n",
    "    q=set(list(clean_list))\n",
    "    if s == q:\n",
    "          print('found all genes supplied')\n",
    "    else:\n",
    "          print('error, some genes missing')\n",
    "          temp3 = [x for x in q if x not in s]\n",
    "          print(str(temp3)+' not found')\n",
    "    \n",
    "    g.close()\n",
    "    print('Gene file written to '+output_dir+\"/\"+clade+'.GeneSequence.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e7dc650",
   "metadata": {},
   "outputs": [],
   "source": [
    "for k, v in dictionary.items():\n",
    "    Gene_roundup(k, v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e11b66b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#sanity check \n",
    "len(numpy.unique(gene['Clade']))\n",
    "#same number of directories as clades\n",
    "genes=list(gene[gene['Clade']=='Int10637_304_324_R_203']['Gene'])\n",
    "#sort(genes)\n",
    "genes.sort()\n",
    "genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b109b19b",
   "metadata": {},
   "source": [
    "Input strategy change, get the Gene names from the alignment and use that to build the FASTA to ensure they are exactly the same "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5d56d5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#generate a dictionary with Clade names as the keys and representative genes as the values \n",
    "dictionary2=dict()\n",
    "gene=gene.astype({'File':'str'})\n",
    "files=numpy.unique(gene['File'])\n",
    "\n",
    "cleanedfiles = [x for x in files if str(x) != 'nan']\n",
    "\n",
    "for file in cleanedfiles:    \n",
    "    genes=[]\n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/NLRCladeFinder/Atha_NLRome/\"+file,'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "            if line.startswith('>'):\n",
    "                genes.append(line.replace('>', '').replace(' \\n', ''))\n",
    "                \n",
    "    \n",
    "    clade=file.split('/')[-1].replace('.afa', '')\n",
    "    dictionary2[clade]=genes\n",
    "\n",
    "dictionary2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4cf4e2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the Col0 genes so I can pull sequences from phytozome \n",
    "all_genes=list(dictionary2.values())\n",
    "flat_list = [item for sublist in all_genes for item in sublist]\n",
    "for gene in flat_list:\n",
    "    if gene.startswith(\"ATHALIANA\"):\n",
    "        col0=gene.replace(\"ATHALIANA_\", '').strip(\"_1\")\n",
    "        print(col0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "149aad20",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Gene roundup sandbox to get the same name as in the alignment file \n",
    "def Gene_roundup_test(clade, gene_list):\n",
    "    #initialize some housekeeping, count being the number of found exons and found a set of gene names \n",
    "    count=0\n",
    "    found=[]\n",
    "    \n",
    "    #get the clade name and initialize the new output directory \n",
    "    output_dir=\"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    if not os.path.exists(output_dir): #this prevents a directory exists error\n",
    "       os.makedirs(output_dir)\n",
    "\n",
    "    # create output Gene Sequences file \n",
    "    g = open(output_dir+\"/\"+clade+'.GeneSequence_test.fa', \"w\")\n",
    "\n",
    "    # open the NLRome CDS file and search for gene names \n",
    "    #if found, write to new fasta file \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/hvNLR/ArabidopsisRENSEQ/GeneSequences/NLRome_Gene_Sequences.respectify.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "            for gene in gene_list:\n",
    "                clean=gene.replace('_T', '_G').replace('-R1', '').replace('_1', '.1').replace('_2', '.2').replace('_3', '.3')\n",
    "                if clean in line:\n",
    "                    count=count+1\n",
    "                    found.append(gene)\n",
    "                    g.writelines('>'+gene+'\\n')\n",
    "                    g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #special treatment for Col-0 because of the full gene names \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/popgen/Col0_GeneSequences.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "                for gene in gene_list:\n",
    "                    if gene.startswith('ATHALIANA'):\n",
    "                        short=gene.replace('ATHALIANA_', '').replace('_1', '').replace('.1', '')\n",
    "                        if short in line: \n",
    "                            count=count+1\n",
    "                            found.append(gene)\n",
    "                            g.writelines('>'+gene+'\\n')\n",
    "                            g.writelines(lines[lines.index(line)+1])\n",
    "    \n",
    "    #sanity check. The found genes should be the same as the input genes. \n",
    "    print('found '+str(count)+' genes')\n",
    "    s=set(found)\n",
    "    q=set(list(gene_list))\n",
    "    if s == q:\n",
    "          print('found all genes supplied')\n",
    "    else:\n",
    "          print('error, some genes missing')\n",
    "          temp3 = [x for x in q if x not in s]\n",
    "          print(str(temp3)+' not found')\n",
    "    \n",
    "    g.close()\n",
    "    print('Gene file written to '+output_dir+\"/\"+clade+'.GeneSequence.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a86f1736",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gene_roundup_test(clade, dictionary2[clade])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "902a3adf",
   "metadata": {},
   "source": [
    "Switch strategy part 2: get the CDS together as single sequences to try not to break palnal2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "126e904c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#input: a clade name, and the list of genes found in the alignment in Daniil's format \n",
    "#output: a fasta file written, with Daniil gene IDs. Exons are concatenated to form the transcript sequence. \n",
    "#update: for col0, use tair10 transcripts \n",
    "def transcript_roundup(clade, gene_list):\n",
    "    #initialize some housekeeping, count being the number of found exons and found a set of gene names \n",
    "    exons=[]\n",
    "    found=[]\n",
    "\n",
    "    #get the clade name and initialize the new output directory \n",
    "    output_dir=\"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    if not os.path.exists(output_dir): #this prevents a directory exists error\n",
    "        os.makedirs(output_dir)\n",
    "\n",
    "    # create output CDS file \n",
    "    g = open(output_dir+\"/\"+clade+'.transcript.fa', \"w\")\n",
    "\n",
    "    for gene in gene_list:\n",
    "        #special treatment for Col-0 because of the full gene names \n",
    "        count=0\n",
    "        if gene.startswith('ATHALIANA'):\n",
    "            #using just the primary transcript\n",
    "            clean=gene.replace('ATHALIANA_', '').replace('_1', '')+'.1'\n",
    "            transcript=[]\n",
    "            line_number=0\n",
    "\n",
    "            with open(\"/global/scratch/users/chandlersutherland/e14/popgen/Col0_CDS_tair10.fa\",'r') as text_file:\n",
    "                lines = text_file.readlines()\n",
    "                for line in lines:\n",
    "                    line_number += 1\n",
    "                    if line.find(clean) != -1: \n",
    "                        transcript.append(lines[line_number].strip('\\n'))\n",
    "                        count += 1\n",
    "                        \n",
    "            found.append(gene)\n",
    "            exons.append([gene, count])\n",
    "            g.writelines('>'+gene+'\\n')\n",
    "            g.writelines(''.join(transcript)+'\\n')\n",
    "         \n",
    "        #everything else, search the NLRome CDS file \n",
    "        else:\n",
    "            clean=gene.replace('_1', '.1').replace('_2', '.2').replace('_3', '.3')\n",
    "            transcript=[]\n",
    "            line_number=0\n",
    "\n",
    "            with open(\"/global/scratch/users/chandlersutherland/e14/hvNLR/ArabidopsisRENSEQ/CDS/NLRome.CDS.fasta\",'r') as text_file:\n",
    "                lines = text_file.readlines()\n",
    "                for line in lines:\n",
    "                    line_number += 1\n",
    "                    if line.find(clean) != -1: \n",
    "                        transcript.append(lines[line_number].strip('\\n'))\n",
    "                        count += 1\n",
    "\n",
    "            found.append(gene)\n",
    "            exons.append([gene, count])\n",
    "            g.writelines('>'+gene+'\\n')\n",
    "            g.writelines(''.join(transcript)+'\\n')\n",
    "\n",
    "    g.close()\n",
    "    \n",
    "    #check that output transcripts matches number of input \n",
    "    s=set(found)\n",
    "    q=set(list(gene_list))\n",
    "    if s == q:\n",
    "          print('found all genes supplied; '+str(len(s))+' genes\\n\\n')\n",
    "    else:\n",
    "          print('error, some genes missing')\n",
    "          temp3 = [x for x in q if x not in s]\n",
    "          print(str(temp3)+' not found')\n",
    "    print('Exons added per gene: ')\n",
    "    print(exons)\n",
    "    print('\\n\\n')\n",
    "    \n",
    "    g.close()\n",
    "    print('Gene file written to '+output_dir+\"/\"+clade+'.transcript.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49b6b883",
   "metadata": {},
   "outputs": [],
   "source": [
    "clade='Int10637_304_324_R_203'\n",
    "gene_list=dictionary2[clade]\n",
    "transcript_roundup(clade, gene_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "820f1c94",
   "metadata": {},
   "outputs": [],
   "source": [
    "for k, v in dictionary2.items():\n",
    "    transcript_roundup(k, v)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b298bb86",
   "metadata": {},
   "source": [
    "It would be helpful to copy the best alignment file into the created clade folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a9385d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in cleanedfiles: \n",
    "    clade=file.split('/')[-1].strip('.afa')\n",
    "    input_file = \"/global/scratch/users/chandlersutherland/e14/NLRCladeFinder/Atha_NLRome/\" + file\n",
    "    output_dir = \"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    os.popen('cp '+input_file+' '+output_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd05037f",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleanedfiles"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca47406e",
   "metadata": {},
   "source": [
    "Similarly for the tree output "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8730498",
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in cleanedfiles: \n",
    "    subfolder=file.split('/')[-2]\n",
    "    clade=file.split('/')[-1].strip('.afa')\n",
    "    input_file = \"/global/scratch/users/chandlersutherland/e14/NLRCladeFinder/Atha_NLRome/RAxML_tree_pbNB-ARC/\" + subfolder+'/RAxML_bipartitionsBranchLabels.'+clade+'.Raxml.out'\n",
    "    output_dir = \"/global/scratch/users/chandlersutherland/e14/popgen/clades/\"+clade\n",
    "    os.popen('cp '+input_file+' '+output_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e853e778",
   "metadata": {},
   "source": [
    "Create a single fasta file with all the NLR proteins in the NLRome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94f96d9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes=gene.loc[:,'Gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e69c3ea6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#does not have to be clade aware, just making a single fasta\n",
    "# create output fasta file \n",
    "g = open('/global/scratch/users/chandlersutherland/e14/popgen/NLRome_proteins.fasta', \"w\")\n",
    "\n",
    "for gene in all_genes:\n",
    "    count=0\n",
    "    protein=[]\n",
    "    line_number=0\n",
    "    \n",
    "    #convert gene name \n",
    "    if gene.startswith('ATHALIANA'):\n",
    "        #using just the primary transcript\n",
    "        clean=gene.replace('ATHALIANA_', '').replace('_1', '')+'.1'\n",
    "    else:\n",
    "        clean=gene.replace('_1', '.1').replace('_2', '.2').replace('_3', '.3')\n",
    "    \n",
    "    #search file with clean gene name \n",
    "    with open(\"/global/scratch/users/chandlersutherland/e14/hvnlr_clades/hvNLR/Athaliana_NLR_Phylogeny/Proteomes/NLRome.aa.fa\",'r') as text_file:\n",
    "        lines = text_file.readlines()\n",
    "        for line in lines:\n",
    "            line_number += 1\n",
    "            if line.find(clean) != -1: \n",
    "                protein.append(lines[line_number].strip('\\n'))\n",
    "                count += 1\n",
    "\n",
    "        found.append(gene)\n",
    "        g.writelines('>'+gene+'\\n')\n",
    "        g.writelines(''.join(protein)+'\\n')\n",
    "\n",
    "g.close()\n",
    "\n",
    "#check that output transcripts matches number of input \n",
    "s=set(found)\n",
    "q=set(list(gene_list))\n",
    "if s == q:\n",
    "      print('found all genes supplied; '+str(len(s))+' genes\\n\\n')\n",
    "else:\n",
    "      print('error, some genes missing')\n",
    "      temp3 = [x for x in q if x not in s]\n",
    "      print(str(temp3)+' not found')\n",
    "        \n",
    "print('\\n\\n')\n",
    "\n",
    "g.close()\n",
    "print('Gene file written to '+'/global/scratch/users/chandlersutherland/e14/popgen/NLRome_proteins.fasta')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
