NLR protein alignments were converted to codon alignments using the `pal2nal.bash` script. Whole genome information was generated using the .vcf files produced by 1001 genomes, subset to those present in the Arabidopsis NLRome using `vcf_thinner.bash`. NLR specific population genetics statistics were calculated with `egglib_nlr.ipyb` and whole genome with `egglib_whole_genome.ipynb`. Domain specific annotations of alignments were generated using `nlr_annotation.ipynb`. Positive selection analysis was done with `hyphy_meme_runner.bash`, `hyphy_fel_runner.bash`, and `fubar_meme_collate.ipynb`. The list of analyzed clades is provided in `clades.txt`. 
