---
title: "Enrichment Plots"
output: html_notebook
---

Adapted from singscore visualization of ranked density reported in Foroutan M, Bhuva D, Lyu R, Horan K, Cursons J, Davis M (2018). “Single sample scoring of molecular phenotypes.” BMC bioinformatics, 19(1), 404. doi: 10.1186/s12859-018-2435-4.  

```{r}
library(tidyverse)
library(ggplot2)
library(openxlsx)
library(ggsignif)
library(ggpubr)
```

```{r}
#load in all gene data 
table <- readxl::read_xlsx(path="C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\all_gene_table.xlsx")
grey <- read.csv("C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E8//Ath_somatic_mutations_gene_level_data.csv")
merged <- merge(grey, table  %>% subset(select=c('Gene', 'HV')), by.x='gene', by.y='Gene', all=TRUE) %>% 
  mutate(HV=recode(HV, `0` = "non-hv", `1`="hv")) %>% dplyr::rename('Gene'='gene')
```

Write a function that takes in a column name, then outputs a ranked list 
```{r}
ranking <- function(table, x){
  filt <- table[,c('Gene', 'HV', x)] 
  filt <- filt[complete.cases(filt), ]
  filt <- filt[order(filt[[x]]),]
  rownames(filt) <- NULL
  hv <- filt %>% filter(HV == 'hv') %>% pull('Gene')
  nonhv <- filt %>% filter(HV == 'non-hv') %>% pull('Gene')
  
  #assign the rank and calculate normalized rank 
  filt$rank <- row.names(filt) %>% as.numeric()
  filt$norm_rank <- filt$rank/nrow(filt)
  
  #find the hv and nonhv rank 
  hv_rank <- filt %>% subset(Gene %in% hv) %>% mutate(set='hv')
  nhv_rank <- filt %>% subset(Gene %in% nonhv) %>% mutate(set='non-hv')
  
  #bind and return ranks
  rank <- rbind(hv_rank, nhv_rank)
  rank <- rank[c('norm_rank', 'set')]
  
  rank
}

exp_rank <- ranking(table, 'log2_TPM')
meth_rank <- ranking(table, "meth_percentage")
te_rank <- ranking(table, 'te_dist')

write.table(exp_rank, 'C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\exp_rank.csv', sep=',')
write.table(meth_rank, 'C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\meth_rank.csv', sep=',')
write.table(te_rank, 'C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\te_rank.csv', sep=',')
```

Methylation enrichment plot:
```{r}
#methylation enrichment plot 
p1 <- ggplot(meth_rank, aes(x = norm_rank, col = set)) +
  	stat_density(aes(y = ..density..), geom = 'line', position = 'identity', linewidth=0.3)

#Define axes 
ymap = c(2.6, -0.1)

#Plot
meth_enrich_p <- p1 + 
  geom_segment(data= meth_rank %>% filter(set=='hv'), aes(
      x= norm_rank,
  		y = ymap[1],
  		xend = norm_rank,
  		yend = ymap[1]+0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
geom_segment(data= meth_rank %>% filter(set=='non-hv'), aes(
      x= norm_rank,
  		y = ymap[2],
  		xend = norm_rank,
  		yend = ymap[2] -0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
  theme_classic()+
  theme(legend.position='none', text=element_text(size=10), 
        axis.title=element_blank())+
  annotate("text", x = 0.22, y = 1.6, label = "*", color='#F8766D', size=3) #greatest p value for hvNLR methylation weighted by GC content is .02, hvnlr is .55 (see permutations.Rmd for calculation)

meth_enrich_p
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\met_enrich.png', plot=meth_enrich_p, dpi='retina', width=2, height=1.5)
```

```{r}
p2 <- ggplot(exp_rank, aes(x = norm_rank, col = set)) +
  	stat_density(aes(y = ..density..), geom = 'line', position = 'identity', linewidth=0.3)

#Define Axes
ymap = c(2.6, -0.1)

exp_enrich_p <- p2 + geom_segment(data= exp_rank %>% filter(set=='hv'), aes(
      x= norm_rank,
  		y = ymap[1],
  		xend = norm_rank,
  		yend = ymap[1]+0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
geom_segment(data= exp_rank %>% filter(set=='non-hv'), aes(
      x= norm_rank,
  		y = ymap[2],
  		xend = norm_rank,
  		yend = ymap[2] -0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
  theme_classic()+
  theme(legend.position='none', text=element_text(size=10), axis.title=element_blank()) +
  annotate("text", x = 0.825, y = 2.1, label = "**", color='#F8766D', size=3) #Greatest sample p value for singscore enrichment is 0.002

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\exp_enrich.png', plot=exp_enrich_p, dpi='retina', width=2, height=1.5)
```

```{r}
p3 <- ggplot(te_rank, aes(x = norm_rank, col = set)) +
  	stat_density(aes(y = ..density..), geom = 'line', position = 'identity', linewidth=0.3)

ymap = c(2.6, -0.1)

te_enrich_p <- p3 + geom_segment(data= te_rank %>% filter(set=='hv'), aes(
      x= norm_rank,
  		y = ymap[1],
  		xend = norm_rank,
  		yend = ymap[1]+0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
geom_segment(data= te_rank %>% filter(set=='non-hv'), aes(
      x= norm_rank,
  		y = ymap[2],
  		xend = norm_rank,
  		yend = ymap[2] -0.3), alpha = 0.75, linewidth=0.3) +
  	scale_colour_manual(values=c('#F8766D', '#00BFC4')) +
  	labs(colour = c('hvNLR', 'nonhvNLR')) +
  theme_classic() +
  theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  annotate("text", x = 0.07, y = 2.37, label = "***", color='#F8766D', size=3) #p value is 0 for hv, see permutations.Rmd

te_enrich_p
ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\te_enrich.png', plot=te_enrich_p, dpi='retina', width=2, height=1.5)
```


