---
title: "generating_table"
author: "Chandler Sutherland"
date: '2022-08-25'
output: html_document
---

```{r}
knitr::opts_knit$set(root.dir = 'C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14')
library(tidyverse)
library(ggplot2)
library(openxlsx)
library(ggsignif)
library(ggpubr)
```

Read in the supplemental table from Chae 2020 and the HV status from Prighozin and Krasileva 2021
```{r}
chae <- readxl::read_xlsx(path="C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\chae_2020_supplemental.xlsx")

clades <- read.table(file='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\Atha_NLRome_GeneTable.txt', header=1) %>% 
  filter(Ecotype=='ATHALIANA') %>% 
  separate(Gene, c(NA, 'Gene'))%>%
  subset(select=c('Gene', 'Clade', 'HV')) %>% 
  distinct()

#merge on Gene name 
table <- merge(chae, clades, by.x='gene', by.y='Gene', all.y=TRUE)

#Manually add information for At4G19050, which is missing from Chae analysis 
index <- which(table$gene == 'AT4G19050')
table[index, 3] = 'Chr4'
table[index, 4] = '10439983'
table[index, 5] = '10444121'
table[index, 6] = '-'
table[index, 7] = 'singleton'
table[index, 18] = 'C' 

table %>% filter(gene=='AT4G11170')
```

QC
```{r}
HV_count <- sum(table$HV == 1)
nHV_count <- sum(table$HV == 0)

paste("There are ", HV_count, "HV genes in my working table and ", nHV_count, " non-HV genes in my working table.")
hvNLR <- table %>% filter(HV=='hv') %>% pull('gene')
nonhvNLR <- table %>% filter(HV=='non-hv') %>% pull('gene') %>% unique()
```

```{r}
#per gene average % CpG methylation 
SRR17281085 <- readxl::read_xlsx(path="C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E14//e14_R//SRR17281085_per_gene_meth_count.xlsx", skip=2, col_names = c('Index', 'Chrom', 'Gene', 'mean_percent_methylation', 'count'))
SRR17281086 <- readxl::read_xlsx(path="C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E14//e14_R//SRR17281086_per_gene_meth_count.xlsx", skip=2, col_names = c('Index', 'Chrom', 'Gene', 'mean_percent_methylation', 'count'))
SRR17281087 <- readxl::read_xlsx(path="C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E14//e14_R//SRR17281087_per_gene_meth_count.xlsx", skip=2, col_names = c('Index', 'Chrom', 'Gene', 'mean_percent_methylation', 'count'))
SRR17281088 <- readxl::read_xlsx(path="C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E14//e14_R//SRR17281088_per_gene_meth_count.xlsx", skip=2, col_names = c('Index', 'Chrom', 'Gene', 'mean_percent_methylation', 'count'))

all_meth <- rbind(SRR17281085, SRR17281086, SRR17281087, SRR17281088) %>%
  group_by(Gene) %>% summarize(meth_percentage=mean(mean_percent_methylation))

all_meth
```


```{r}
#TPM 
TPM <- read.table("C:/Users/chand/Box Sync/Krasileva_Lab/Research/chandler/Krasileva Lab/E14/e14_R/tpm_matrix.csv", header=TRUE, sep=",", row.names=2)%>% dplyr::select(-X)

TPM_df <- TPM %>% mutate('mean' = rowMeans(TPM[1:4])) 
all_expression <- merge(TPM_df, all_meth, by.x=0, by.y='Gene', all=TRUE)
all_expression <- all_expression %>% mutate(log2_TPM = log2(all_expression$mean+1))

#have to manually set the unmappable genes, since they just look like 0 
unmappable <- c('AT1G58807', 'AT1G58848', 'AT1G59124', 'AT1G59218')

all_expression$log2_TPM[all_expression$Row.names %in% unmappable] <- NA   
all_expression$mean[all_expression$Row.names %in% unmappable] <- NA 
```


```{r}
#TE distance
TE_table <- read.table(file='C:/Users/chand/Box Sync/Krasileva_Lab/Research/chandler/Krasileva Lab/E14/e14_R/Atha_genes_with_TE_dist.tsv', header=1) %>% distinct()
#converte TE distance to kb 
TE <- merge(TE_table, all_expression, by.x = 'Gene', by.y='Row.names', all=TRUE) %>%
  mutate(te_dist = te_dist/1000)
```


```{r}
#clean and concentrate 
merged <- merge(TE, table, by.x='Gene', by.y='gene', all=TRUE)
all_table <- merged %>% subset(select=c('Gene', 'HV', 'log2_TPM', 'meth_percentage', 'te_dist')) %>%
  mutate(HV=recode(HV, `0` = "non-hv", `1`="hv"))

all_table$HV[is.na(all_table$HV)]<-'all_genes'
all_table$HV <- factor(all_table$HV , levels=c("all_genes", "non-hv", "hv"))

all_table
HV_count <- sum(all_table$HV == 'hv')
nHV_count <- sum(all_table$HV == 'non-hv')

paste("There are ", HV_count, "HV genes in my working table and ", nHV_count, " non-HV genes in my working table.")
```
```{r}
#popgen stats from egglib generated on NLRome coding sequences as described in pi_calculation.ipynb 
NLR_popgen <- read_tsv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//egglib_results.tsv") %>% 
  filter(domain=='cds')%>%
  mutate(Pi_by_dna=Pi_raw/dna_aln)%>%
  mutate(Pi_by_lseff=Pi_raw/lseff) %>%
  subset(select=c('Gene', 'D','Pi_by_dna', 'Pi_by_lseff'))
  
NLR_popgen 

#rest of the genome from vcf as described in whole_genome_popgen.ipynb
whole_genome <- read_csv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//cds_egglib_all.csv") %>% 
  subset(select=c('gene', 'Pi_by_dna', 'Pi_by_lseff', 'D')) %>%
  rename('Gene'='gene') %>% distinct()

whole_genome 

#replace NLRs in whole genome with more accurate long read versions 
NLRs <- NLR_popgen %>% pull('Gene') %>% unique()
no_NLRs <- whole_genome %>% filter(!Gene %in% NLRs)

popgen_table <- rbind(no_NLRs, NLR_popgen) %>% arrange(Gene)
popgen_table
final_table <- merge(all_table, popgen_table) %>% distinct()
HV_count <- sum(final_table$HV == 'hv')
nHV_count <- sum(final_table$HV == 'non-hv')

final_table %>% filter(HV!='all_genes') %>% distinct() %>% group_by(Gene) %>% filter(n()>1)
paste("There are ", HV_count, "HV genes in my working table and ", nHV_count, " non-HV genes in my working table.")
```


```{r}
write.xlsx(final_table, 'C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\all_gene_table.xlsx')
```

