---
title: "popgen_plots"
author: "Chandler Sutherland"
date: "2023-08-15"
output: html_document
---

Purpose of script: Plot population genetics statistics of hv and non-hvNLRs  

Copyright (c) Chandler Sutherland
Email: chandlersutherland@berkeley.edu

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(ggplot2)
library(ggsignif)
library(ggpubr)
library(ggbeeswarm)
require(tidyverse)
```
Gather and clean the necessary datasets 
```{r}
#read in col0_popgen_stats, generated by the pi_calculation script and cleaned by the popgen_table script 
col_popgen <- read_tsv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//col0_popgen_stats.tsv")

#add sample sizes to the label 
to_plot <- col_popgen %>% group_by(HV) %>%
  mutate(n=n()) %>%
  mutate(label = paste0(HV,"\n", "n=",n)) %>%
  mutate(PiNPiS = PiN/PiS) 

labels <- to_plot$label %>% unique()
to_plot$label <- factor(to_plot$label, levels=c(labels[1], labels[2]))

#read in per domain stats generated by pi_calculation script 
col_per_domain <- read_tsv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//col0_per_domain_stats.tsv") %>% mutate(PiNPiS = PiN/PiS)
col_per_domain[c('PiNPiS')][sapply(col_per_domain[c('PiNPiS')], is.infinite)] <- NA

#add the entire coding sequence information
cds <- to_plot %>% subset(select=c('Clade', 'Gene', 'HV', 'name', 'D', 'Pi', 'PiN', 'PiS', 'PiNPiS')) %>% mutate('domain'='CDS')
all <- rbind(col_per_domain %>% subset(select=-c(start, stop)), cds) %>%
  mutate(domain=recode(domain, 'cc' = "CC", 'tir'="TIR", 'nbarc'='NBARC', 'lrr'='LRR'))

all$domain <- factor(all$domain, levels=c('CDS', 'CC', 'TIR', 'NBARC', 'LRR'))
all$HV <- factor(all$HV, levels=c('non-hv', 'hv'))

#read in the sliding window analysis generated by pi_calculation script
col_slider <- read_csv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//pi_slider_100.csv")
col_slider <- col_slider %>% 
  mutate(nuc_midpoint = (start+50)*3) %>% #convert from codon to CDS nucleotide length 
  mutate(nuc_length=length*3) %>% 
  mutate(normal=nuc_midpoint/nuc_length) %>% #normalize by length to aggregate groups together 
  rename(Clade=clade)

title <- col_per_domain %>% subset(select=c('Clade', 'Gene', 'HV', 'name')) %>% unique()
named_slider <- col_slider %>% merge(title, all.y=TRUE)


#read in Grey's mutation data 
grey <- read.csv("C://Users//chand//Box Sync//Krasileva_Lab//Research//chandler//Krasileva Lab//E8//Ath_somatic_mutations_gene_level_data.csv")
table <- readxl::read_xlsx(path="C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\all_gene_table.xlsx")
merged <- merge(grey, table  %>% subset(select=c('Gene', 'HV')), by.x='gene', by.y='Gene', all=TRUE) %>% 
  mutate(HV=recode(HV, `0` = "non-hv", `1`="hv"))

merged$HV[is.na(merged$HV)]<-'all_genes'
merged$HV <- factor(merged$HV , levels=c("all_genes", "non-hv", "hv"))

NLR_only <- merged %>% filter(HV != 'all_genes') %>%
  mutate(per_bp=obs_mutations/obs_length)

grey_variants <- read_csv("//wsl.localhost//Ubuntu//home//chandlersutherland//scratch//raw_variants.csv", col_names=c('Chrom', 'position', 'type'))
```

# Figure 3: 
Per domain plots 
```{r}
pi_exp <- expression(pi)
p7 <- ggplot(all,
       aes(x=domain, y=Pi, fill=HV))+
  geom_boxplot(lwd=0.25, outlier.size=0.1)+
    scale_fill_manual(values=c('#00BFC4', '#F8766D'))+
  theme_classic()+
  ylab(pi_exp)+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())
  
p7 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pi_domain.png', plot=p7, dpi='retina', width=2.2, height=1.5)

compare_means(Pi~HV, all %>% filter(domain=='CDS'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='NBARC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='CC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='TIR'), method='wilcox.test', paired=FALSE)$p/5
compare_means(Pi~HV, all %>% filter(domain=='LRR'), method='wilcox.test', paired=FALSE)$p/5

compare_means(Pi~domain, all, method='wilcox.test')
compare_means(Pi~domain, all %>% filter(HV=='hv'), method='wilcox.test')
compare_means(Pi~domain, all %>% filter(HV=='non-hv'), method='wilcox.test')

kruskal.test(Pi ~ domain, data = all)
kruskal.test(Pi ~ domain, data = all%>%filter(HV=='hv'))
kruskal.test(Pi ~ domain, data = all%>%filter(HV=='non-hv'))
```
```{r}
p10 <- ggplot(all,
       aes(x=domain, y=D))+
   geom_boxplot(lwd=0.25, aes(fill=HV), outlier.size = 0.1)+
    scale_fill_manual(values=c('#00BFC4', '#F8766D'))+
  theme_classic()+
  ylab("Tajima's D")+
    theme(legend.position='none', text=element_text(size=10), axis.title.x = element_blank())

  
p10 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_domain.png', plot=p10, dpi='retina', width=2.2, height=1.5)

compare_means(D~HV, all %>% filter(domain=='NBARC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(D~HV, all %>% filter(domain=='CC'), method='wilcox.test', paired=FALSE)$p/5
compare_means(D~HV, all %>% filter(domain=='TIR'), method='wilcox.test', paired=FALSE)$p/5
compare_means(D~HV, all %>% filter(domain=='LRR'), method='wilcox.test', paired=FALSE)$p/5
compare_means(D~HV, all %>% filter(domain=='CDS'), method='wilcox.test', paired=FALSE)$p/5

compare_means(D~domain, all, method='wilcox.test')
compare_means(D~domain, all %>% filter(HV=='hv'), method='wilcox.test')
compare_means(D~domain, all %>% filter(HV=='non-hv'), method='wilcox.test')

kruskal.test(D ~ domain, data = all)
kruskal.test(D ~ domain, data = all%>%filter(HV=='hv'))
kruskal.test(D ~ domain, data = all%>%filter(HV=='non-hv'))
```


Sliding window plots 
```{r}
p8 <- ggplot(data=named_slider, aes(x=normal, y=Pi, color=HV))+
  geom_smooth(lwd=0.5)+
  theme_classic()+
  xlab('Normalized Nucleotide Position')+
  ylab(pi_exp)+
  #geom_hline(yintercept=0.005)+
    theme(text=element_text(size=10), 
        legend.text=element_text(size=10), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))

p8 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pi_window.png', plot=p8, dpi='retina', width=3.2, height=1.5)

p9 <- ggplot(data=named_slider, aes(x=normal, y=D, color=HV))+
  geom_smooth(lwd=0.5)+
  theme_classic()+
  xlab('Normalized Nucleotide Position')+
  ylab("Tajima's D")+
  #geom_hline(yintercept=-0.8251341)+
    theme(text=element_text(size=10), 
        legend.text=element_text(size=10), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))

p9 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_window.png', plot=p9, dpi='retina', width=3.2, height=1.5)
```

# Figure 4: 
PiN and PiS plots 
```{r}
pi_long <- to_plot %>% subset(select=c('name', 'PiS', 'PiN', 'HV')) %>% pivot_longer(
    cols = c(PiS, PiN), 
    names_to = "pi_type",
    values_to = "value") %>%
  mutate(pi_type=recode(pi_type, 
                        "PiS" = "{pi[S]}", 
                        "PiN" = "{pi[N]}"))



p13 <- pi_long %>%
  ggplot(aes(x=HV, y=value, fill=HV))+
  geom_violin(lwd=0.25)+
    geom_quasirandom(alpha=0.3, size=0.5)+
   geom_signif(comparisons=list(c('hv', 'non-hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25, y=c(0.08))+
  ylim(0,0.1)+
    scale_fill_manual(values=c('#F8766D', '#00BFC4'))+
  theme_classic()+
    theme(legend.position='none', text=element_text(size=10), axis.title=element_blank())+
  facet_grid(~pi_type, labeller=label_parsed)

p13

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pisandpin.png', plot=p13, dpi='retina', width=2.5, height=2)
```


PiN/PiS plot
```{r}
pis=expression(pi['S'])
pin=expression(pi['N'])
pis_equals_pin=expression(pi['s']~'='~pi['n'])

pis_piN <- to_plot %>% ungroup() %>% mutate(clean_name = case_when(name == 'RPP13' ~ 'RPP13', 
                                                        name == 'RPP7' ~ 'RPP7', 
                                                        name == 'DSC1' ~ 'DSC1', 
                                                        name == 'RPP8' ~ 'RPP8', 
                                                        name == 'RPS5' ~ 'RPS5')) %>%
  ggplot(aes(x=PiS, y=PiN, fill=HV))+
    geom_point(aes(color=HV), size=0.5)+
  geom_abline(slope=1, intercept=0, color='darkgrey', linetype='dashed')+
  ylim(0,0.08)+
  xlim(0,0.08)+
  scale_fill_manual(values = c("hv" = "#F8766D",
                               "non-hv" = '#00BFC4'))+
  geom_text(aes(label=clean_name), size=3)+
  xlab(pis)+
  ylab(pin)+
  theme_classic()+
  theme(text=element_text(size=11), 
        legend.text=element_text(size=11), 
        legend.title = element_blank(), 
        legend.key.size = unit(.2, 'cm'), 
        legend.key.height=unit(.2, 'cm'),
        legend.key.width = unit(.2, 'cm'))+
  annotate(geom="text", label=pis_equals_pin, x=0.06, y=0.0625, angle=40, vjust=-1, color='darkgrey', size=3)

pis_piN 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\pis_pin.png', plot=pis_piN, dpi='retina', width=5.4, height=3.6)
```

Mutation plots 
```{r}
mut_prob_plot <- NLR_only %>%
  ggplot(aes(x=HV, y=Mutation.Probability.Score, fill=HV))+
  geom_boxplot(lwd=0.25, outlier.size=0.1)+
  scale_fill_manual(values=c('#00BFC4', '#F8766D'))+
  theme_classic()+
  geom_signif(comparisons=list(c('non-hv', 'hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  theme(legend.position='none', text=element_text(size=10))+
  ylim(0,1.2e-4)+
  scale_y_continuous(labels = scales::scientific)+
  ylab("Mutation Probability")+
  xlab('')

mut_prob_plot 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\mutprob.png', plot=mut_prob_plot, dpi='retina', width=1.75, height=2)

mut_obs_plot <- NLR_only %>%
  ggplot(aes(x=HV, y=per_bp, fill=HV))+
  geom_boxplot(lwd=0.25, outlier.size=0.1)+
  scale_fill_manual(values=c('#00BFC4', '#F8766D'))+
  theme_classic()+
  geom_signif(comparisons=list(c('non-hv', 'hv')), 
               map_signif_level = TRUE, test=wilcox.test, textsize=2, size=0.25)+
  theme(legend.position='none', text=element_text(size=10))+
  ylab("de novo Mutations/bp")+
  xlab('')

mut_obs_plot

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\mut_obs.png', plot=mut_obs_plot, dpi='retina',  width=1.25, height=2)

compare_means(Mutation.Probability.Score~HV, NLR_only, method="wilcox.test")$p/2
compare_means(per_bp~HV, NLR_only, method="wilcox.test")$p/2
```
# Figure 5: 

AT5G43730
Chr5:17560178..17564434 forward
CDS only: 88-2635  

cc: 82-186 on CDS
nbarc: 472-1311
LRR: 1534-2472
```{r}
#make annotation table 
gene <- c('AT5G43730', 'AT5G43730', 'AT5G43730')
domain <- c('CC', 'NBARC', 'LRR')
start <- c(82, 472, 1534)
stop <- c(186, 1311, 2472)
coords <-data.frame(gene, domain, start, stop)

coords$domain <- factor(coords$domain, levels=c('CC', "NBARC", 'LRR'))
plotter <- grey_variants %>% 
  filter(Chrom == 5 & position >= 17560178+88 & position <=17564434-1621) %>%
  mutate_at('position', as.numeric) %>% 
  mutate(rel_pos=position-(17560178+88))

mutation <- ggplot(plotter)+
  ylim(0, 3)+
  theme_classic()+
  xlab('Nucleotide Position')+
  geom_rect(data=coords, aes(xmin=start, xmax=stop,
                             ymin=0, ymax=0, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
  geom_point(aes(x=rel_pos, y=0.2, fill=type), shape=25, size=2)+
  scale_fill_manual(values=c('black', 'white'))+
  labs(fill='Mutation Type', color='Domain')+
  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(), 
        axis.line.y=element_blank())

mutation 

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\mut_case.png', plot=mutation, dpi='retina', width=5, height=3)
```

AT5G43740
Chr5:17564696..17569148 forward
CDS start 1313 after 
CDS end 550 bp before 

#cc: 78-162 on CDS
#nbarc: 468-1308
#LRR: 1530-2490
```{r}
gene <- c('AT5G43740', 'AT5G43740', 'AT5G43740')
domain <- c('CC', 'NBARC', 'LRR')
start <- c(78, 468, 1530)
stop <- c(162, 1308, 2490)
coords2 <-data.frame(gene, domain, start, stop)
coords2$domain <- factor(coords$domain, levels=c('CC', "NBARC", 'LRR'))
plotter2 <- grey_variants %>% 
  filter(Chrom == 5 & position >= (17564696+1313) & position <=(17569148-550)) %>%
  mutate_at('position', as.numeric) %>% 
  mutate(rel_pos=position-(17564696+1313))

mutation2 <- ggplot(plotter2)+
  ylim(0, 3)+
  theme_classic()+
  xlab('Nucleotide Position')+
  geom_rect(data=coords2, aes(xmin=start, xmax=stop,
                             ymin=0, ymax=0, color=domain),size=5)+
   scale_color_manual(values=c('#00997F', '#00BADE', '#9661FF'))+
  geom_point(aes(x=rel_pos, y=0.2, fill=type), shape=25, size=2)+
  scale_fill_manual(values=c('black', 'white'))+
  labs(fill='Mutation Type', color='Domain')+
  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(), 
        axis.line.y=element_blank())


ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\mut_case2.png', plot=mutation2, dpi='retina', width=5, height=3)

mutation2
```

```{r}
Pi_AT5G43730 <- named_slider %>% filter(Gene=='AT5G43730') %>% ggplot() + 
    geom_point(aes(x=nuc_midpoint, y=Pi), size=1)+
    geom_line(aes(x=nuc_midpoint, y=Pi))+
 geom_vline(xintercept=82, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=186, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=472, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1311, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1534, linetype='dashed', color='#9661FF')+
  geom_vline(xintercept=2472, linetype='dashed', color='#9661FF')+
  ylim(0, 0.1)+
  xlab('300bp Window Midpoint')+
  ylab(pi_exp)+
  xlim(0, 2547)+
  theme_classic()+
  theme(legend.position='none', 
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        axis.line.x=element_blank())
Pi_AT5G43730

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\Pi_30.png', plot=Pi_AT5G43730, dpi='retina', width=4, height=1)

Pi_AT5G43740 <- named_slider %>% filter(Gene=='AT5G43740') %>% ggplot() + 
    geom_point(aes(x=nuc_midpoint, y=Pi), size=1)+
    geom_line(aes(x=nuc_midpoint, y=Pi))+
  geom_vline(xintercept=78, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=162, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=468, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1308, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1530, linetype='dashed', color='#9661FF')+
  geom_vline(xintercept=2490, linetype='dashed', color='#9661FF')+
  ylim(0, 0.1)+
  xlim(0, 2589)+
  ylab(pi_exp)+
  theme_classic()+
  theme(legend.position='none',
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        axis.line.x=element_blank())

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\Pi_40.png', plot=Pi_AT5G43740, dpi='retina', width=4, height=1)

Pi_AT5G43740
```


```{r}
D_AT5G43730 <- named_slider %>% filter(Gene=='AT5G43730') %>% filter(!is.na(D))%>%
  ggplot() + 
    geom_point(aes(x=nuc_midpoint, y=D))+
    geom_line(aes(x=nuc_midpoint, y=D))+
  geom_vline(xintercept=82, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=186, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=472, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1311, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1534, linetype='dashed', color='#9661FF')+
  geom_vline(xintercept=2472, linetype='dashed', color='#9661FF')+
  ylim(-3, 3)+
  ylab("D")+
  xlim(0, 2547)+
  theme_classic()+
  theme(legend.position='none',
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        axis.line.x=element_blank())

D_AT5G43730

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_30.png', plot=D_AT5G43730, dpi='retina', width=4, height=1)

D_AT5G43740 <- named_slider %>% filter(Gene=='AT5G43740') %>% ggplot()+
    geom_point(aes(x=nuc_midpoint, y=D))+
    geom_line(aes(x=nuc_midpoint, y=D))+
  geom_vline(xintercept=78, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=162, linetype='dashed', color='#00997F')+
  geom_vline(xintercept=468, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1308, linetype='dashed', color='#00BADE')+
  geom_vline(xintercept=1530, linetype='dashed', color='#9661FF')+
  geom_vline(xintercept=2490, linetype='dashed', color='#9661FF')+
  ylim(-3, 3)+
  ylab("D")+
  xlim(0, 2589)+
  theme_classic()+
  theme(legend.position='none',
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        axis.line.x=element_blank())

D_AT5G43740

ggsave(filename='C:\\Users\\chand\\Box Sync\\Krasileva_Lab\\Research\\chandler\\Krasileva Lab\\E14\\e14_R\\final_figures\\D_40.png', plot=D_AT5G43740, dpi='retina', width=4, height=1)
```